app/assets/javascripts/models/ui/reverb_retirement.coffee

```diff
@@ -0,0 +1,49 @@
+#= require util/js.cookies
+#= require util/metrics
+
+ns = @edsc.models.ui
+data = @edsc.models.data
+
+ns.ReverbRetirement = do (ko,
+                          ajax = @edsc.util.xhr.ajax,
+                          config=@edsc.models.data.config) ->
+  class ReverbRetirement
+    constructor: ->
+
+    referrerIsReverb: () =>
+      console.log "Checking referrer: " + document.referrer
+      console.log "Checking referrer against ENV value: " + edsc.config.reverb_url
+      referrer = if document.referrer then document.referrer.match(/:\/\/(.[^/]+)/)[1] else false
+      reverb = if edsc.config.reverb_url then edsc.config.reverb_url.match(/:\/\/(.[^/]+)/)[1] else false
+      return referrer == reverb && referrer
+
+    returnToReverb: (source) =>
+      Cookies.set('ReadyForReverbRetirement', 'false', { expires: 90 })
+      data['type'] = 'reverb_redirect'
+      data['data'] = 'back_to_reverb'
+      data['other_data'] = "{source: '" + source + "'}"
+      ajax
+        data: JSON.parse(JSON.stringify(data))
+        dataType: 'json'
+        url: "/metrics"
+        method: 'post'
+        success: (data) ->
+          console.log data
+      window.location.replace("https://" + document.referrer.match(/:\/\/(.[^/]+)/)[1])
```

Comment:
Rename this file to ```app/assets/javascripts/models/ui/reverb_retirement.coffee.erb``` and use ```edsc.config.reverb_url``` explicitly. (We talked about this in gotomeeting. It is _that_ issue.)

Reply:
Which line are you referring to in regards to using edsc.config.reverb_url explictely?  From what I can tell it already does that on line 16.

I am not sure why the file needs to be renamed as well.  :\

Reply:
This line. Depends on how you wanna make the change. It could be something _**like**_ ```window.location.replace("<%=ENV['reverb_root'] || Rails.configuration.services['edsc'][Rails.env]['reverb_root']%>")``` and rename it as an ```.erb``` file. Or you could say ```window.location.replace(edsc.config.reverb_url)```

Reply:
In the gotomeeting, you had said that this line was fine as it was.  It was one of the first concerns we talked about.  You had a concern that it was using the referrer and were worried about a security concern; when I pointed out that the referrer was filtered by a regex, you said (paraphrased) "Oh - that's okay then." and we moved on.

What changed?  :\

Reply:
To add - I'll give in on this one and say 'uncle' if you really want it - it's just that we agreed to do it this way in the gotomeeting (since it has a regex filter it was deemed 'safe').  If you reaaaally want it though, I'll do it - we need to wrap this all up.  Let me know.  :)

Reply:
I thought it was something else - like you set the referrer using the ENV first then use it in the redirect. In fact, there is no code to set this ```document.referrer``` so it is subject to others to set it to something else. That's the reason that I want what inside of ```window.location.replace``` to be a hard-coded ENV value.

Reply:
I will update it, however - it's harmless as it is.  The regex makes it so.  Even if someone spoofs a referrer in, what could they do with it?  The best they could do is trigger the soft redirect link when it shouldn't come up.  This wasn't really a security concern.

However, I'll make the change, incoming in 5, 4, 3, 2...